nodelet_topic_tools/include/nodelet_topic_tools/nodelet_lazy.h

```diff
@@ -144,28 +164,23 @@ class NodeletLazy: public nodelet::Nodelet
     if (lazy_)
     {
       boost::mutex::scoped_lock lock(connection_mutex_);
-      for (size_t i = 0; i < publishers_.size(); i++)
+      if (hasSubscribers())
```

Comment:
This is a very small and cosmetic change, but the diff is large because of one `if` block getting into the method. I recommend hiding whitespace changes.

Now this method can be overridden by the child class to also check for the `image_transport::Publisher`s.